### PR TITLE
refactor v2 order

### PR DESCRIPTION
**Pull request by marktoda**
Separate cosigned and unsigned orders more clearly for easier integration

### CODE REVIEW

File: src/builder/V2DutchOrderBuilder.ts [hunk 1]
```diff
@@ -110,7 +103,7 @@ export class V2DutchOrderBuilder extends OrderBuilder {
       output.startAmount.gte(output.endAmount),
       `startAmount must be greater than endAmount: ${output.startAmount.toString()}`
     );
-    this.info.outputs.push(output);
+    this.info.outputs?.push(output);
```
Comment (zhongeric) [inline src/builder/V2DutchOrderBuilder.ts:106]: shouldn't need ? here if initialized in constructor? unless diff types in inherited classes from UnsignedXV2Order

File: src/builder/V2DutchOrderBuilder.ts [hunk 2]
```diff
@@ -278,7 +275,7 @@ export class V2DutchOrderBuilder extends OrderBuilder {
     );
     this.info.cosignerData.outputOverrides.forEach((override, idx) => {
       invariant(
-        override.lte(this.info.outputs[idx].startAmount),
+        override.lte(this.info.outputs![idx].startAmount),
```
Comment (zhongeric) [inline src/builder/V2DutchOrderBuilder.ts:278]: add check that outputOverrides.length == info.outputs.length

**Review by zhongeric — APPROVED** [review]

**Comment by marktoda** [comment]
will do comments in followup ty

File: src/builder/V2DutchOrderBuilder.ts (hunk 1 above)
Comment (marktoda) [reply src/builder/V2DutchOrderBuilder.ts:106]: fixed in #119

**Pull request merged**